mysql-plugin/src/test/java/co/cask/mysql/MysqlPluginTestBase.java

```diff
@@ -0,0 +1,240 @@
+                      DATAPIPELINE_ARTIFACT_ID,
+                      Sets.newHashSet(hypersql), Driver.class);
+
+    TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
+
+    connectionUrl = "jdbc:mysql://" + BASE_PROPS.get("host") + ":" +
+      BASE_PROPS.get("port") + "/" + BASE_PROPS.get("database");
+    Connection conn = createConnection();
+    createTestTables(conn);
+    prepareTestData(conn);
+  }
+
+  protected static void createTestTables(Connection conn) throws SQLException {
+    try (Statement stmt = conn.createStatement()) {
+      // create a table that the action will truncate at the end of the run
+      stmt.execute("CREATE TABLE dbActionTest (x int, day varchar(10))");
+      // create a table that the action will truncate at the end of the run
+      stmt.execute("CREATE TABLE postActionTest (x int, day varchar(10))");
+
+      // note that the tables need quotation marks around them; otherwise, hsql creates them in upper case
+      stmt.execute("CREATE TABLE my_table" +
+                     "(" +
+                     "ID INT NOT NULL, " +
+                     "NAME VARCHAR(40) NOT NULL, " +
+                     "SCORE DOUBLE, " +
+                     "GRADUATED BOOLEAN, " +
+                     "NOT_IMPORTED VARCHAR(30), " +
+                     "TINY TINYINT, " +
+                     "SMALL SMALLINT, " +
+                     "MEDIUMINT_COL MEDIUMINT, " +
+                     "BIG BIGINT, " +
+                     "FLOAT_COL FLOAT, " +
+                     "REAL_COL REAL, " +
+                     "NUMERIC_COL NUMERIC(10, 2), " +
+                     "DECIMAL_COL DECIMAL(10, 2), " +
+                     "BIT_COL BIT, " +
+                     "DATE_COL DATE, " +
+                     "TIME_COL TIME, " +
+                     "TIMESTAMP_COL TIMESTAMP(3), " +
+                     "YEAR_COL YEAR, " +
+                     "TEXT_COL TEXT," +
+                     "TINYTEXT_COL TINYTEXT," +
+                     "MEDIUMTEXT_COL MEDIUMTEXT," +
+                     "LONGTEXT_COL LONGTEXT," +
+                     "CHAR_COL CHAR(100)," +
+                     "BINARY_COL BINARY(100)," +
+                     "VARBINARY_COL VARBINARY(20)," +
+                     "TINYBLOB_COL TINYBLOB, " +
+                     "BLOB_COL BLOB(100), " +
+                     "MEDIUMBLOB_COL MEDIUMBLOB, " +
+                     "LONGBLOB_COL LONGBLOB " +
+                     ")");
+      stmt.execute("CREATE TABLE MY_DEST_TABLE AS " +
+                     "SELECT * FROM my_table");
+      stmt.execute("CREATE TABLE your_table AS " +
+                     "SELECT * FROM my_table");
+    }
+  }
+
+  protected static void prepareTestData(Connection conn) throws SQLException {
+
+    try (Statement stmt = conn.createStatement()) {
+      stmt.execute("insert into dbActionTest values (1, '1970-01-01')");
+      stmt.execute("insert into postActionTest values (1, '1970-01-01')");
+    }
+
+    try (
+      PreparedStatement pStmt1 =
```

Comment:
statements should be in try-with-resource blocks

Reply:
actually they are